src/librbd/ImageCtx.cc

```diff
@@ -809,6 +811,20 @@ class ThreadPoolSingleton : public ThreadPool {
     }
   }
 
+  void ImageCtx::clear_pending_completions() {
+    Mutex::Locker l(completed_reqs_lock);
+    ldout(cct, 10) << "clear pending AioCompletion: count="
+                   << completed_reqs.size() << dendl;
+
+    for (xlist<AioCompletion*>::iterator it = completed_reqs.begin();
+         !it.end(); ++it) {
+      ldout(cct, 10) << "clear pending AioCompletion: " << *it << dendl;
+      (*it)->set_event_notify(false);
```

Comment:
Sorry, I don't get your point about why it not help. After set to false, AioCompletion complete won't access ImageCtx again, so that we can avoid client refer to a destructed Image